### PR TITLE
Peer sockets

### PR DESCRIPTION
**Pull request by BraydenLangley**

### CODE REVIEW

File: src/tokenator.js [hunk 1]
```diff
@@ -65,13 +141,17 @@ class Tokenator {
       throw e
     }
 
-    // Notify the token management server about the new token
+    const hmac = await createHmac({ data: Buffer.from(JSON.stringify(message.body)), protocolID: `${message.messageBox} peerserv`, keyID: '1', counterparty: message.recipient })
```
Comment (ty-everett) [inline src/tokenator.js:144]:
```suggestion
    const hmac = await createHmac({ data: Buffer.from(JSON.stringify(message.body)), protocolID: [0, 'PeerServ'], keyID: '1', counterparty: message.recipient })
```

File: src/tokenator.js [hunk 2]
```diff
@@ -32,17 +33,92 @@ class Tokenator {
...
+   * @param {object} obj all params given in an object
+   * @param {string} obj.message The message contents to send
+   * @param {string} obj.messageBox The messageBox the message should be sent to depending on the protocol being used
+   * @param {string} obj.recipient The identityKey of the intended recipient
+   */
+  async sendLiveMessage({ message, messageBox, recipient }) {
+    await this.initializeConnection()
+
+    // Send the message to the recipients message box first
+    const result = await this.sendMessage({
```
Comment (ty-everett) [inline src/tokenator.js:96]: Why do this first? The sockets are supposed to be live, and if the user is offline, they don't care about having a delay. Should the sockets be first?

File: src/tokenator.js [hunk 3]
```diff
@@ -32,17 +33,92 @@ class Tokenator {
...
+   * Start listening on your "public" message room
+   * Anyone can send you a message here
+   * @param {object} obj - all params given in an object
+   * @param {function} obj.onMessage - onMessage handler function
+   * @param {string} obj.messageBox - name of messageBox to listen on 
+   */
+  async listenForLiveMessages({ onMessage, messageBox }) {
+    await this.initializeConnection(messageBox)
+
+    this.authriteClient.on('sendMessage', async (payload) => {
```
Comment (ty-everett) [inline src/tokenator.js:79]: Is this callback specific to the message box we are listening on? We should ensure it only registers messages for the correct message box. For example, if I use this multiple times, I suspect that each callback would get both sets of messages.

If I want to bind box A to function A, and box B to function B, we should ensure that function A never gets called with box B messages, and function A never gets called with box B messages.

File: src/tokenator.js [hunk 4]
```diff
@@ -32,17 +33,92 @@ class Tokenator {
...
+        this.myIdentityKey = this.authriteClient.clientPublicKey
+      }
+    }
+
+    if (!this.authriteClient.socket) {
+      await this.authriteClient.connect(this.peerServHost)
+
+      // Wait until the connection response has been sent
+      // TODO: Support a callback or similar function from the authrite side.
+      await setTimeout(async () => {
```
Comment (ty-everett) [inline src/tokenator.js:59]: Support for this at the Authrite layer should be created before merging this code.

**Comment by BraydenLangley** [comment]
All issues should be addressed. Please review updates

**Review by ty-everett — APPROVED** [review]

**Pull request merged**